### DISCUSSION TITLE
AMR is compatible with VDA 5050 Version 1.012 according to manufacturer. What Version do I use in the commadapter-VDA5050?

### DISCUSSION THREAD

**Discussion by antichecker** [openTCS/opentcs#152] [Q&A] [answered]
The manufacturer of the AMR told me that it's compatible with VDA Version 1.012. Do I use Version 1.1 or 2.0 now in the commadapter?

**Comment by swltr** [comment, marked as answer]
@antichecker VDA5050 version 1.012 is not compatible with either 1.1 or 2.0, unfortunately.

**Comment by antichecker** [comment]
Thank you for your answer. Do you know if it's possible to work around this incompatibility with the libvda5050++ implementation?

**Reply by swltr** [reply]
@antichecker I assume you mean https://git.openlogisticsfoundation.org/silicon-economy/libraries/vda5050/libvda5050pp. Yes, it could be used to implement VDA5050 1.1 or 2.0 on the vehicle side. Actually, I would recommend using this existing library instead of creating your own vehicle-side VDA5050 implementation. A prerequisite is that the AMR provides some way to integrate a C++ library, of course, but I would expect most AMRs to do that.

**Reply by antichecker** [reply]
@swltr 
Yes I meant this, thanks.
The libvda5050++ README.md states that it's implementing the VDA 5050 standard version 2.0.0. 
I assume using version 2.0 of the commadapter-VDA5050 on the openTCS side is required to allow successful communication?

**Reply by swltr** [reply]
@antichecker The VDA5050 1.1 and 2.0 implementations are kept on separate branches there, so you do have a choice. I would still use 2.0 for new projects, though.

Yes, if you use 2.0 on the vehicle side, you'll have to use 2.0 on the openTCS side, too, since the two versions are not compatible with each other.
